base_miner/UCF/trainer/trainer.py

```diff
@@ -0,0 +1,448 @@
+# This script was adapted from the DeepfakeBench training code,
+# originally authored by Zhiyuan Yan (zhiyuanyan@link.cuhk.edu.cn)
+
+# Original: https://github.com/SCLBD/DeepfakeBench/blob/main/training/train.py
+
+import os
+import sys
+current_file_path = os.path.abspath(__file__)
+parent_dir = os.path.dirname(os.path.dirname(current_file_path))
+project_root_dir = os.path.dirname(parent_dir)
+sys.path.append(parent_dir)
+sys.path.append(project_root_dir)
```

Comment:
[Suggestion] To wrap up my 1billion comments on these hacky path management snippets - if removing these is a really annoying lift that requires reformatting all local imports, we can leave it as-is for now and address in a subsequent PR